Fix: do not use the profile name, url or port in game save files

SlySven opened the pull request:
#### Brief overview of PR changes/additions
This is counterproductive as the data is supposed to be stored in the base directory of the profile and changes made in the "Connect profile" dialog get overridden by the game save.

#### Motivation for adding to Mudlet
I nearly got banned for multi-playing on a Server because I was working on a separate issue related to multiple auto loading profiles and two of the profiles I normally used had been edited to set the URL to `localhost` but when the profiles were autoloaded they used the stored settings and promptly BOTH connected to the Game server - waking up its anti- multi-playing detection!

As the profile name is not now being stored within the profile save file there is no need to backup and then restore it when installing a package, so code to do that and the same for the character name and login password is also removed from `Host::installPackage(...)`.

SlySven committed:
Fix: do not store/load profile name, url or port in game save files

This is counterproductive as the data is supposed to be stored in the base
directory of the profile and changes made in the "Connect profile" dialog
get overridden by the game save.

I nearly got banned for multi-playing on a Server because I was working on
a separate issue related to multiple auto loading profiles and two of the
profiles I normally used had been edited to set the URL to `localhost` but
when the profiles were autoloaded they used the stored settings and
promptly BOTH connected to the Game server - waking up its anti-
multi-playing detection!

As the profile name is not now being stored within the profile save file
there is no need to backup and then restore it when installing a package,
so code to do that and the same for the character name and login password
is also removed from `Host::installPackage(...)`.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 reviewed (commented):
I agree it's trouble, but what about when you want to 'save profile as' and import it on another system?

A better solution would be for code to differentiate the context it's running in: just loading, or importing.

SlySven commented:
> I agree it's trouble, but what about when you want to 'save profile as' and import it on another system?

Well the profile will gain the name of the directory it is named to be in won't it? Or do you mean "keep the details in the save file and not rely on it except in *some condition* I am not quite clear about yet"?

vadi2 commented:
> Well the profile will gain the name of the directory it is named to be in won't it?

How will that work with this:

>  'save profile as' and import it on another system?

There will be no directory, just an XML file.

SlySven commented:
> > Well the profile will gain the name of the directory it is named to be in won't it?
> 
> How will that work with this:
> 
> > 'save profile as' and import it on another system?
> 
> There will be no directory, just an XML file.

But the user will have to put that XML save file into a `./current/` sub-directory for a new profile off of the Mudlet `./profile/` one and whatever they call that directory will be the profile name (hopefully it will be a *valid* name...!)

vadi2 commented:
Not necessarily, it should be simple to use so they should be able to import that file from wherever it is.

SlySven commented:
Well that is where they **have to** have it located to use it currently. Mudlet is not portable yet remember! :laughing:

vadi2 commented:
That's not what I mean. Try doing what I mentioned - 'Save Profile As' and then 'Import' it'll 'kind of' import the Mudlet profile. Doesn't work great right now, but that's why for this one usecase we need these parameters.

SlySven committed:
Revise: revert removal of storing some details in save but don't load them

Rather read them and leave them in some "backup" variables for future code
to use if needed.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
> That's not what I mean. Try doing what I mentioned - 'Save Profile As' and then 'Import' it'll 'kind of' import the Mudlet profile. Doesn't work great right now, but that's why for this one usecase we need these parameters.

Okay then, I've partly reverted things so that the particular details are now still written to the game save file but they are read into three new, otherwise unused variables:
* `(QString) Host::mBackupHostName`
* `(QString) Host::mBackupUrl`
* `(int) Host::mBackupPort`

so that they are available for the hypothetical "import" code to use but which does not overwrite the "proper" data in normal usage.

BTW When you say "Save Profile As" you mean the button in the Editor don't you?

vadi2 commented:
Yup
